mlflow/utils/version.py

```diff
@@ -0,0 +1,28 @@
+import importlib.metadata
+import operator
+
+from packaging.version import Version
+
+operators = {
+    ">": operator.gt,
+    ">=": operator.ge,
+    "<": operator.lt,
+    "<=": operator.le,
+    "==": operator.eq,
+    "!=": operator.ne,
+}
+
+
+def compare_package_version(package: str, version: str, cmp: str) -> bool:
+    """
+    Compare the version of a package with a given version.
+    :param package: The name of the package.
+    :param version: The version to compare with.
+    :param cmp: The comparison operator.
+    :return: True if the version of the package satisfies the comparison.
+    """
+
+    if cmp not in operators:
+        raise ValueError(f"Invalid comparison operator: {cmp}")
+    op = operators[cmp]
+    return op(Version(importlib.metadata.version(package)), Version(version))
```

Comment:
I'd make this function return a `Version` instance.

Reply:
in this case, the caller will need to import `Version` and to compare with the target version themselves?

Reply:
maybe i'll just remove this function and inline in gcs repo, what do you think?

Reply:
>  i'll just remove this function and inline in gcs repo

Can you share what the diff would look like?

Reply:
https://github.com/mlflow/mlflow/pull/10124#discussion_r1377258518 looks good to me.